centreon/www/install/steps/process/migrateCredentialsToVault.php

```diff
@@ -153,9 +154,14 @@ function migrateDatabaseCredentials(
  */
 function updateConfigFilesWithVaultPath($vaultPath): void
 {
+    $featuresFileContent = file_get_contents(__DIR__ . '/../../../../config/features.json');
+    $featureFlagManager = new FeatureFlags(false, $featuresFileContent);
+
     updateCentreonConfPhpFile($vaultPath);
-    updateCentreonConfPmFile($vaultPath);
-    updateDatabaseYamlFile($vaultPath);
+    if ($featureFlagManager->isEnabled('vault_broker')) {
+        updateCentreonConfPmFile($vaultPath);
+        updateDatabaseYamlFile($vaultPath);
+    }
```

Comment:
This one is for the installation process but the method also exists in vault-functions (used by migration script).